src/test/java/ru/training/at/hw1/CalculatorSubtractTest.java

```diff
@@ -0,0 +1,16 @@
+package ru.training.at.hw1;
+
+import com.epam.tat.module4.Calculator;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class CalculatorSubtractTest {
+
+    @Test (groups = {"sumAndSubtractTests", "allTests"},
+           dataProviderClass = DataProviders.class, dataProvider = "subtractData")
+    public void calculatorSubtractTest(long a, long b, long result) {
+        Calculator calculator = new Calculator();
```

Comment:
Why do create instance of **calculator** in each test? May be it will be easily  create `before` and `after` hooks?

Reply:
I tried it with the hooks you mentioned - ended up with Null Pointer Exception(
Will try again

As for build problems - could you please suggest what I should do? Looks like I put my calculator-1.0.jar file in some other directory than it is in your project, but what can be done here? Please advice.